addons/mass_mailing/static/src/js/mailing_m2o_filter.js

```diff
@@ -0,0 +1,158 @@
+ *
+ * In edit mode, it will allow to save the latest configured domain
+ * in form of the favorite filter, or to remove the store filters.
+ *
+ */
+export class FieldMany2OneMailingFilter extends Many2OneField {
+    setup() {
+        super.setup();
+        this.notification = useService("notification");
+        onMounted(this._updateFilterIcons);
+        onPatched(this._updateFilterIcons);
+        useExternalListener(document, "show.bs.dropdown", this.onFilterDropdownShown);
+    }
+
+    /**
+     * Updates the 'Add to favorite' / 'Remove' icons' visibility based on the
+     * current state, and shows the custom message when no filter is available
+     * for the selected model.
+     *
+     * The filter can be saved if one of those conditions is matched:
+     * - No favorite filter is currently set
+     * - User emptied the input
+     * - User changed the domain when favorite filter is set
+     * - The input is currently being edited, known by the "this.state.isFloating" variable
+     *
+     * @private
+     */
+    _updateFilterIcons() {
+        const el = document.querySelector('.o_mass_mailing_filter_container');
+        if (!el || this.props.readonly) {
+            return;
+        }
+        const filterCount = this.props.record.data.mailing_filter_count;
+        el.querySelector('.o_mass_mailing_no_filter').classList.toggle('d-none', filterCount);
+        document.querySelectorAll('.o_field_many2one_selection > .o_input_dropdown')[1].classList.toggle('d-none', !filterCount);
+        // By default, domains in recordData are in string format, but adding / removing a leaf from domain widget converts
+        // value into object, so we use 'Domain' class to convert them in same (string) format, allowing proper comparison.
+        const recordDomain = new Domain(this.props.record.data[this.props.domain_field] || []).toString();
+        const filterDomain = new Domain(this.props.record.data.mailing_filter_domain || []).toString();
+        el.classList.toggle('d-none', recordDomain === '[]');
+        const canSaveFilter = !this.props.record.data.mailing_filter_id
+            || !document.querySelector(`input#${this.props.model_field}`).value.length
+            || this.state.isFloating
+            || filterDomain !== recordDomain;
+        el.querySelector('.o_mass_mailing_save_filter_container').classList.toggle('d-none', !canSaveFilter);
+        el.querySelector('.o_mass_mailing_remove_filter').classList.toggle('d-none', canSaveFilter);
+    }
+
+    // HANDLERS
+
+    /**
+     * On opening the drop-down, remove existing input value (if any) and set
+     * focus on input
+     *
+     * @param {event} ev
+     */
+    onFilterDropdownShown(ev) {
+        const filterInput = ev.currentTarget.querySelector('input.o_mass_mailing_filter_name');
+        filterInput.value = '';
+        setTimeout(() => filterInput.focus({ preventScroll: true }), 100);
```

Comment:
I tend to dislike those arbitrary setTimeout ^^

If we had a small component for the dropdown, we could use onMounted to auto-focus.